editor/create_dialog.h

```diff
@@ -64,13 +64,23 @@ class CreateDialog : public ConfirmationDialog {
 	HashMap<String, TreeItem *> search_options_types;
 	HashMap<String, String> custom_type_parents;
 	HashMap<String, int> custom_type_indices;
-	List<StringName> type_list;
+	// Pair of <type_name, search_keywords>
```

Comment:
I'd keep the comment since `StringName` and `PackedStringArray` don't clearly describe what the Pair contains. Could make it a struct to name these fields but figured it's fine like this if it's just search keywords.

Reply:
I would make it a struct then, so you can name the items, and I think we will move to that in the future anyway because I am seeing a lot of stuff we are doing during search that we can precompute (at least partially):

 - _is_type_preferred
 - _add_type: parent class
 - _configure_search_option_item: metadata, icon, etc

Reply:
Changed to a struct.